component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/emailotp/EmailOTPAuthenticator.java

```diff
@@ -301,6 +262,110 @@ protected void processAuthenticationResponse(HttpServletRequest request, HttpSer
     }
 
     /**
+     * Process the EmailOTP Flow
+     * @param request the request
+     * @param response response
+     * @param email value of the email to send otp
+     * @param username the username
+     * @param queryParams the queryParams
+     * @param retryParam the retryParams
+     * @param context the authentication context
+     * @throws AuthenticationFailedException
+     */
+    private void processEmailOTPFlow(HttpServletRequest request,
+                                     HttpServletResponse response, String email, String username, String queryParams,
+                                     String retryParam, AuthenticationContext context)
+            throws AuthenticationFailedException {
+        Map<String, String> authenticatorProperties = context.getAuthenticatorProperties();
+        Map<String, String> emailOTPParameters = getAuthenticatorConfig().getParameterMap();
+        try {
+            if (!context.isRetrying() || (context.isRetrying()
+                    && StringUtils.isEmpty(request.getParameter(EmailOTPAuthenticatorConstants.RESEND)))
+                    || (context.isRetrying()
+                    && Boolean.parseBoolean(request.getParameter(EmailOTPAuthenticatorConstants.RESEND)))) {
+
+                OneTimePassword token = new OneTimePassword();
+                String secret = OneTimePassword.getRandomNumber(EmailOTPAuthenticatorConstants.SECRET_KEY_LENGTH);
+                String myToken = token.generateToken(secret, "" + EmailOTPAuthenticatorConstants.NUMBER_BASE
+                        , EmailOTPAuthenticatorConstants.NUMBER_DIGIT);
+                context.setProperty(EmailOTPAuthenticatorConstants.OTP_TOKEN, myToken);
+                if (authenticatorProperties != null) {
```

Comment:
Again no readability. avoid nested if blocks. 

Eg:- If there are exception cases do those first and return from that at the top.